src/mono/mono/utils/mono-dl-posix.c

```diff
@@ -136,10 +136,10 @@ mono_dl_convert_flags (int flags)
 
 #ifdef ENABLE_NETCORE
 	// Specifying both will default to LOCAL
-	if (flags & MONO_DL_LOCAL)
-		lflags |= RTLD_LOCAL;
-	else if (flags & MONO_DL_GLOBAL)
+	if (flags & MONO_DL_GLOBAL && !(flags & MONO_DL_LOCAL))
 		lflags |= RTLD_GLOBAL;
+	else 
+		lflags |= RTLD_LOCAL;
```

Comment:
Commenting just to note that this is a significant behavioral change to ensure we default to RTLD_LOCAL on both OSX and Linux. This was the intention from the start and was discussed beforehand, but is worth bringing attention to.